spring-ai-core/src/main/java/org/springframework/ai/vectorstore/VectorStore.java

```diff
@@ -34,4 +38,57 @@ default void accept(List<Document> documents) {
 	 */
 	List<Document> similaritySearch(String query, int k, double threshold);
 
+	/**
+	 * Retrieve documents with similar to the query content. You can limit your vector
+	 * search based on {@link Document} metadata. Spring AI lets you attach metadata
+	 * key-value pairs to Documents stored in teh {@link VectorStore}, and specify filter
+	 * expressions when you query the them.
+	 *
+	 * Searches with metadata filters retrieve exactly the number of nearest-neighbor
+	 * results that match the filters. For most cases, the search latency will be even
+	 * lower than unfiltered searches.
+	 *
+	 * NOTE: This filter expressions have different syntax for each Vector Store, making
+	 * this method not portable across different {@link VectorStore} implementations. For
+	 * portable implementations use
+	 * {@link VectorStore#similaritySearch(String, int, double, org.springframework.ai.vectorstore.filter.Filter.Expression)}
+	 * instead. The {@link Filter.Expression} is portable across most Spring AI vector
+	 * store implementation. in combination with {@link FilterExpressionTextParser} or
+	 * {@link FilterExpressionBuilder}.
+	 * @param query The query to send, it will be converted to an embedding based on the
+	 * configuration of the vector store.
+	 * @param k the top 'k' similar results
+	 * @param threshold the lower bound of the similarity score
+	 * @param vectorSpecificFilterExpression vector store specific metadata filters.
+	 * @return similar documents that match the requested threshold and filter.
+	 */
+	default List<Document> similaritySearch(String query, int k, double threshold,
+			String vectorSpecificFilterExpression) {
```

Comment:
this could  change to be the antl4 string and we remove the ability to pass down vendor specific strings, or we expose that under a different method name.